server/leafnode.go

```diff
@@ -2527,7 +2527,11 @@ func (s *Server) leafNodeFinishConnectProcess(c *client) {
 	c.mu.Unlock()
 
 	// Make sure we register with the account here.
-	c.registerWithAccount(acc)
+	if err := c.registerWithAccount(acc); err != nil {
+		c.Errorf("Registering leaf with account %s resulted in error: %v", acc.Name, err)
+		c.closeConnection(ProtocolViolation)
```

Comment:
Should it not be more something like: `MaxAccountConnectionsExceeded` here?

Reply:
That's what I had initially. I guess it depends. if we assume that ErrTooManyAccountConnections is the only error this function will ever return, yes.

Reply:
I don't understand.. which function are you referring to? `leafNodeFinishConnectProcessing` does not return any error..